### PR TITLE
Support keras non-stateful metrics

### PR DESCRIPTION
**Pull request by yangw1234**
Support mutiple non-stateful keras metrics and user defined metrics

### CODE REVIEW

File: pyzoo/test/zoo/tfpark/test_tfpark_model.py [hunk 1]
```diff
@@ -0,0 +1,288 @@
...
+from test.zoo.pipeline.utils.test_utils import ZooTestCase
+import tensorflow as tf
+import numpy as np
+
+from zoo.tfpark import KerasModel, TFDataset
+
+
+class TestTFPark(ZooTestCase):
+
+    def create_model(self):
```
Comment (jason-dai) [inline pyzoo/test/zoo/tfpark/test_tfpark_model.py]: Also need to test models with multiple inputs and/or outputs

File: zoo/src/main/scala/com/intel/analytics/zoo/pipeline/api/net/TFTrainingHelper.scala [hunk 2]
```diff
@@ -194,6 +194,22 @@ class IdentityCriterion extends AbstractCriterion[Activity, Activity, Float]() {
   }
 }
 
+class NonStatefulMetrics(idx: Int, name: String) extends ValidationMethod[Float] {
```
Comment (jason-dai) [inline zoo/src/main/scala/com/intel/analytics/zoo/pipeline/api/net/TFTrainingHelper.scala]: `StatelessMetric` or `KerasStatelessMetric`

File: pyzoo/zoo/pipeline/api/net.py [hunk 3]
```diff
@@ -481,17 +492,25 @@ def to_sample(t):
             return Sample.from_ndarray(nest.flatten(t), [np.array([0.0])])
 
         sample_rdd = data.map(to_sample)
-        if val_outputs is not None and val_labels is not None:
-            if self.dataset.val_rdd is not None:
-                val_rdd = self.dataset.val_rdd.map(to_sample)
+        has_val = val_outputs is not None and val_labels is not None
+        has_metrics = metrics_tensors is not None
+        has_validation = has_val or has_metrics
+        if has_validation:
```
Comment (jason-dai) [inline pyzoo/zoo/pipeline/api/net.py]: can one have both metrics and validation?

File: pyzoo/zoo/pipeline/api/net.py [hunk 4]
```diff
@@ -361,11 +361,17 @@ def __init__(self, val_method, output_length, target_length):
                            val_method, output_length, target_length)
 
 
+class NonStatefulMetrics(JavaValue):
```
Comment (jason-dai) [inline pyzoo/zoo/pipeline/api/net.py]: `StatelessMetric' or `KerasStatelessMetric`

File: pyzoo/zoo/pipeline/api/net.py [hunk 5]
```diff
@@ -553,37 +572,53 @@ def from_keras(cls, keras_model, dataset, val_spilt=0.0, **kwargs):
...
-            val_outputs = None
-            val_labels = None
-            bigdl_val_methods = None
+        val_outputs = None
+        val_labels = None
+        bigdl_val_methods = None
+        metrics_tensors = None
+        metrics_names = None
+        if keras_model.metrics and (new_dataset.val_rdd is not None or val_spilt != 0.0):
+            metrics_tensors = keras_model.metrics_tensors
```
Comment (jason-dai) [inline pyzoo/zoo/pipeline/api/net.py]: shall we perform mapping to validation method first?

**Comment by yangw1234** [comment]
keras metrics is a little more complicated than expected and the api is changed in tf 1.13.1. So I will work on this after we upgrade tensorflow to 1.13.1.

**Comment by yangw1234** [comment]
unit tests will be add in another pr

**Comment by jason-dai** [comment]
Do we have to support stateful metrics at this moment?

**Comment by yangw1234** [comment]
done in other pr

**Pull request closed without merging**